parlai/agents/transformer/modules/decoder.py

```diff
@@ -38,13 +188,21 @@ class TransformerDecoder(nn.Module):
     :param int n_positions: Size of the position embeddings matrix.
     """
 
+    @dataclass
+    class Manifest(TComponent.Manifest):
+        layer: ComponentSpec[TransformerDecoderLayer] = ComponentSpec(
+            TransformerDecoderLayer, TransformerDecoderLayer.Manifest()
```

Comment:
I guess one thing I'm confused about is the relationship between `klass` and `manifest`, e.g., is the following legit?
```
ComponentSpec(
            TransformerDecoderLayer, CustomTransformerDecoderLayer.Manifest()
)
```

Reply:
I guess you've also mentioned in TODO, but it's cleaner to have a 1-1 mapping between `klass` and `manifest`, is that what you have in mind?

Reply:
So yeah, one thing you could do would be to create a new Manifest:
```
@dataclass
class CustomTransformerDecoderLayer(TransformerDecoderLayer.Manifest):
  # override some default values

ComponentSpec(
  TransformerDecoderLayer,
  CustomTransformerDecoderLayer(),
)
```

So I think the value in not having a 1-1 mapping between `klass` and `manifest` is reducing the need to subclass.

But since the default case is a 1-1 mapping, I'd like to make that case less verbose (which is what that comment is about).

Reply:
That'd be a more accurate name, yes :)